File: eAnimalShelter/eAnimalShelter.Services/Database/Animal.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using eAnimalShelter.Model.Enums;


namespace eAnimalShelter.Services.Database
{
    public class Animal
    {
        [Key]
        public int AnimalId { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;
        public int SpeciesId { get; set; }
        [ForeignKey("SpeciesId")]
        public AnimalSpecies Species { get; set; } = null!;
        public int BreedId { get; set; }
         [ForeignKey("BreedId")]
        public AnimalBreed Breed { get; set; } = null!;
        public AnimalGender Gender { get; set; } 
        public DateTime BirthDate { get; set; }
        public string Description { get; set; } = string.Empty;
        public string PersonalityDescription { get; set; } = string.Empty;
        public string HealthStatus { get; set; } = string.Empty;
        public bool IsVaccinated { get; set; }
        public string? MedicalNotes { get; set; }
        public DateTime? ArrivalDate { get; set; }
        public AnimalStatus AdoptionStatus { get; set; }
        public int CreatedBy { get; set; }
         [ForeignKey("CreatedBy")]
        public User CreatedByUser { get; set; } = null!;

        public ICollection<AnimalImage> Images { get; set; } = new List<AnimalImage>();
        public ICollection<AdoptionRequest> AdoptionRequests { get; set; } = new List<AdoptionRequest>();
        
    }


}

File: eAnimalShelter/UI/eanimalshelter_desktop/lib/models/animal.dart
import 'animal_image.dart';

class Animal {
  int animalId;
  String name;

  String? speciesName;
  String? breedName;

  int gender;
  int adoptionStatus;

  String? description;
  String? personalityDescription;
  String? healthStatus;
  bool isVaccinated;
  String? medicalNotes;

  int? speciesId;
  int? breedId;

  DateTime? birthDate;
  DateTime? arrivalDate;

  List<AnimalImage> images;

  Animal({
    required this.animalId,
    required this.name,
    this.speciesName,
    this.breedName,
    required this.gender,
    required this.adoptionStatus,
    required this.description,
    required this.personalityDescription,
    required this.healthStatus,
    this.medicalNotes,
    required this.isVaccinated,
    this.speciesId,
    this.breedId,
    this.birthDate,
    this.arrivalDate,
    required this.images,
  });

  factory Animal.fromJson(
    Map<String, dynamic> json,
  ) {
    return Animal(
      animalId: json["animalId"],
      name: json["name"] ?? "",
      speciesName: json["speciesName"],
      breedName: json["breedName"],
      gender: json["gender"] ?? 0,
      adoptionStatus:
          json["adoptionStatus"] ?? 0,
      description:
          json["description"],
      personalityDescription:
          json["personalityDescription"],
      healthStatus:
          json["healthStatus"],
      medicalNotes:
          json["medicalNotes"],
      isVaccinated:
          json["isVaccinated"] ?? false,
      speciesId: json["speciesId"],
      breedId: json["breedId"],

      birthDate: json["birthDate"] != null
          ? DateTime.parse(json["birthDate"])
          : null,

      arrivalDate: json["arrivalDate"] != null
          ? DateTime.parse(json["arrivalDate"])
          : null,
      images:
          (json["images"] as List?)
                  ?.map(
                    (e) =>
                        AnimalImage.fromJson(e),
                  )
                  .toList() ??
              [],
    );
  }
}